gerrychain/tree.py

```diff
@@ -811,6 +885,89 @@ def bipartition_tree_random(
         return choice(possible_cuts).subset
 
 
+def epsilon_tree_bipartition(
+    graph: nx.Graph,
+    parts: Sequence,
+    pop_target: Union[float, int],
+    pop_col: str,
+    epsilon: float,
+    node_repeats: int = 1,
+    method: Callable = partial(bipartition_tree, max_attempts=10000),
+) -> Dict:
+    """
+    Uses :func:`~gerrychain.tree.bipartition_tree` to partition a tree into
+    two parts of population ``pop_target`` (within ``epsilon``).
+
+    :param graph: The graph
+    :type graph: nx.Graph
+    :param parts: Iterable of part labels (like ``[0,1,2]`` or ``range(4)``)
+    :type parts: Sequence
+    :param pop_target: Target population for each part of the partition
+    :type pop_target: Union[float, int]
+    :param pop_col: Node attribute key holding population data
+    :type pop_col: str
+    :param epsilon: How far (as a percentage of ``pop_target``) from ``pop_target`` the parts
+        of the partition can be
+    :type epsilon: float
+    :param node_repeats: Parameter for :func:`~gerrychain.tree_methods.bipartition_tree` to use.
+        Defaluts to 1.
+    :type node_repeats: int, optional
+    :param method: The partition method to use. Defaults to
+        `partial(bipartition_tree, max_attempts=10000)`.
+    :type method: Callable, optional
+
+    :returns: New assignments for the nodes of ``graph``.
+    :rtype: dict
+    """
+    if len(parts) != 2:
+        raise ValueError(
+            "This function only supports bipartitioning. Please ensure that there"
+            + " are exactly 2 parts in the parts list."
+        )
+
+    flips = {}
+    remaining_nodes = graph.node_indices
+
+    lb_pop = pop_target * (1 - epsilon)
+    ub_pop = pop_target * (1 + epsilon)
+    check_pop = lambda x: lb_pop <= x <= ub_pop
+
+    nodes = method(
+        graph.subgraph(remaining_nodes),
+        pop_col=pop_col,
+        pop_target=pop_target,
+        epsilon=epsilon,
+        node_repeats=node_repeats,
+        one_sided_cut=False,
+    )
+
+    if nodes is None:
+        raise BalanceError()
+
+    part_pop = 0
+    for node in nodes:
+        flips[node] = parts[-2]
+        part_pop += graph.nodes[node][pop_col]
+
+    if not check_pop(part_pop):
+        raise PopulationBalanceError()
+
+    remaining_nodes -= nodes
+
+    # All of the remaining nodes go in the last part
+    part_pop = 0
+    for node in remaining_nodes:
+        flips[node] = parts[-1]
+        part_pop += graph.nodes[node][pop_col]
+
+    if not check_pop(part_pop):
+        raise PopulationBalanceError()
+
+    return flips
+
+
+# TODO: Move these recursive partition functions to their own module. They are not
```

Comment:
keep comment?

Reply:
I would like to keep this for now. We don't have a running TODO.md file quite yet, and this will be easier to track in the coming months. We should really make a full TODO file soon, though